core/arch/arm/plat-imx/conf.mk

```diff
@@ -30,122 +30,130 @@ mx7-flavorlist = \
-
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx6qsabresd mx6dlsabresd \
+	mx6dlsabrelite mx6dhmbedge mx6dlhmbedge))
+CFG_DDR_SIZE ?= 0x40000000
+CFG_NS_ENTRY_ADDR ?= 0x12000000
+endif
 
-# i.MX6UL/ULL specific config
-ifneq (,$(filter y, $(CFG_MX6UL) $(CFG_MX6ULL)))
-include core/arch/arm/cpu/cortex-a7.mk
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx6shmbedge))
+CFG_DDR_SIZE ?= 0x40000000
+CFG_NS_ENTRY_ADDR ?= 0x12000000
+endif
 
-$(call force,CFG_MX6,y)
-$(call force,CFG_SECURE_TIME_SOURCE_REE,y)
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx6qsabrelite mx6dlsabrelite))
+CFG_DDR_SIZE ?= 0x40000000
+CFG_NS_ENTRY_ADDR ?= 0x12000000
+CFG_UART_BASE ?= UART2_BASE
+endif
 
-CFG_DDR_TEETZ_RESERVED_START ?= 0x9E000000
-CFG_TZDRAM_START ?= $(CFG_DDR_TEETZ_RESERVED_START)
-CFG_TZDRAM_SIZE ?= 0x01E00000
-CFG_SHMEM_START ?= ($(CFG_TZDRAM_START) + $(CFG_TZDRAM_SIZE))
-CFG_SHMEM_SIZE ?= 0x00200000
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx6sxsabreauto))
+CFG_DDR_SIZE ?= 0x80000000
+CFG_NS_ENTRY_ADDR ?= 0x80800000
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx6ulevk mx6ullevk))
+CFG_DDR_SIZE ?= 0x20000000
+CFG_NS_ENTRY_ADDR ?= 0x80800000
+endif
 
 # i.MX6 Solo/SoloX/DualLite/Dual/Quad specific config
 ifeq ($(filter y, $(CFG_MX6Q) $(CFG_MX6D) $(CFG_MX6DL) $(CFG_MX6S) \
       $(CFG_MX6SX)), y)
 include core/arch/arm/cpu/cortex-a9.mk
 
-$(call force,CFG_MX6,y)
 $(call force,CFG_PL310,y)
-$(call force,CFG_SECURE_TIME_SOURCE_REE,y)
 
 CFG_PL310_LOCKED ?= y
-CFG_BOOT_SYNC_CPU ?= y
-CFG_BOOT_SECONDARY_REQUEST ?= y
 CFG_ENABLE_SCTLR_RR ?= y
 endif
 
-# i.MX6 Solo/DualLite/Dual/Quad specific config
 ifeq ($(filter y, $(CFG_MX6Q) $(CFG_MX6D) $(CFG_MX6DL) $(CFG_MX6S)), y)
-CFG_TZDRAM_START ?= 0x4E000000
-CFG_TZDRAM_SIZE ?= 0x01F00000
-CFG_SHMEM_START ?= ($(CFG_TZDRAM_START) + $(CFG_TZDRAM_SIZE))
-CFG_SHMEM_SIZE ?= 0x00100000
+CFG_DRAM_BASE ?= 0x10000000
 endif
 
-# i.MX6 SoloX specific config
-ifeq ($(filter y, $(CFG_MX6SX)), y)
-CFG_TZDRAM_START ?= (0x80000000 + $(CFG_DDR_SIZE) - 0x02000000)
-CFG_TZDRAM_SIZE ?= 0x01e00000
-CFG_SHMEM_START ?= ($(CFG_TZDRAM_START) + $(CFG_TZDRAM_SIZE))
-CFG_SHMEM_SIZE ?= 0x00200000
+ifneq (,$(filter y, $(CFG_MX6UL) $(CFG_MX6ULL) $(CFG_MX6SX)))
+CFG_DRAM_BASE ?= 0x80000000
 endif
 
 ifeq ($(filter y, $(CFG_MX7)), y)
-include core/arch/arm/cpu/cortex-a7.mk
+CFG_INIT_CNTVOFF ?= y
+CFG_DRAM_BASE ?= 0x80000000
+endif
+
+ifneq (,$(filter y, $(CFG_MX6) $(CFG_MX7)))
+$(call force,CFG_GENERIC_BOOT,y)
+$(call force,CFG_GIC,y)
+$(call force,CFG_IMX_UART,y)
+$(call force,CFG_PM_STUBS,y)
+$(call force,CFG_WITH_SOFTWARE_PRNG,y)
 
-$(call force,CFG_SECURE_TIME_SOURCE_REE,y)
+CFG_BOOT_SYNC_CPU ?= n
 CFG_BOOT_SECONDARY_REQUEST ?= y
-CFG_INIT_CNTVOFF ?= y
+CFG_CRYPTO_SIZE_OPTIMIZATION ?= n
+CFG_DT ?= y
+CFG_PAGEABLE_ADDR ?= 0
+CFG_PSCI_ARM32 ?= y
+CFG_SECURE_TIME_SOURCE_REE ?= y
+CFG_UART_BASE ?= UART1_BASE
+CFG_WITH_STACK_CANARIES ?= y
 
-CFG_TZDRAM_START ?= (0x80000000 + $(CFG_DDR_SIZE) - 0x02000000)
+CFG_TZDRAM_START ?= ($(CFG_DRAM_BASE) - 0x02000000 + $(CFG_DDR_SIZE))
```

Comment:
i do not really like the hardcoded size at 32M.
you should probably introduce a CFG like  CFG_TEE_RESERVED_SIZE
and then adjust CFG_TZDRAM_SIZE accordingly

Reply:
If people want to use other base and size, they could pass CFG_TZDRAM_START and CFG_TZDRAM_SIZE, right?

Reply:
sure it can be used that way.